src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -236,4 +227,26 @@ private static boolean checkIsAllLinesAreWhitespace(String[] lines, int lineFrom
         }
         return result;
     }
+
+    /**
+     * Calculates the left curly corresponding to the block to be checked.
+     *
+     * @param ast a {@code DetailAST} value
+     * @return the left curly corresponding to the block to be checked
+     */
+    private static DetailAST findLeftCurly(DetailAST ast) {
+        DetailAST leftCurly;
+        if (ast.findFirstToken(TokenTypes.SLIST) == null) {
+            leftCurly = ast.findFirstToken(TokenTypes.LCURLY);
+        }
+        else {
+            leftCurly = ast.findFirstToken(TokenTypes.SLIST);
+        }
+        if (ast.getType() == TokenTypes.LITERAL_CASE
+                && ast.getNextSibling() != null
+                && "{".equals(ast.getNextSibling().getFirstChild().getText())) {
```

Comment:
@romani yeah ! 
consider:

    case 1: ; 
    case 2: { } 

In both examples the case token would have its next sibling as a `slist` ast. In both examples this `slist` ast would have it's first child of type `slist` ast. The first one's child would have `:` as text. The child of the second `slist` ast one would have '{'. If I would have used `ast.getNextSibling.findFirstToken(TokenTypes.SLIST)` here and returned that then later `hasText(leftCurly)` in `visitToken` would generate a NullPointerException.